### PR TITLE
 RDM-10041 Creating CaseLocation complex type

### PR DESCRIPTION
**Pull request by Abi-Laver**
https://tools.hmcts.net/jira/browse/RDM-10041

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by danlysiak — APPROVED** [review]
👍

File: src/test/java/uk/gov/hmcts/ccd/domain/types/RegionValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    private static final JsonNodeFactory NODE_FACTORY = JsonNodeFactory.instance;
+    private static final String FIELD_ID = "TEST_FIELD_ID";
+
+    @Mock
+    private BaseType regionBaseType;
+
+    @Mock
+    private CaseDefinitionRepository definitionRepository;
+
+    private TextValidator validator;
```
Comment (RebeccaBaker) [inline src/test/java/uk/gov/hmcts/ccd/domain/types/RegionValidatorTest.java:39]: this needs to be RegionValidator not TextValidator

File: src/test/java/uk/gov/hmcts/ccd/domain/types/BaseLocationValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    private static final JsonNodeFactory NODE_FACTORY = JsonNodeFactory.instance;
+    private static final String FIELD_ID = "TEST_FIELD_ID";
+
+    @Mock
+    private BaseType baseLocationBaseType;
+
+    @Mock
+    private CaseDefinitionRepository definitionRepository;
+
+    private TextValidator validator;
```
Comment (RebeccaBaker) [inline src/test/java/uk/gov/hmcts/ccd/domain/types/BaseLocationValidatorTest.java:39]: this needs to be BaseLocationValidator not TextValidator

**Review by danlysiak — APPROVED** [review]
👍

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**